Test Utilities/Doubles/State.swift

```diff
@@ -0,0 +1,23 @@
+@testable import PusherChatkit
+
+public struct TestState {
```

Comment:
This shouldn't live in the `Doubles` group/directory because it isn't a test double, this is test data itself.  

Its generally bad practice to put test data into a common place because it "hides" the data away from the test.  Developers should be able to read the test data at the test site to determine whats going on.  

Having looked a little closer at the use of these values I've determined that its used almost exclusively to fill in blanks that aren't important for a given test. 

Therefore you can simply remove this class/file if you: 

- replace all use of `TestState.user`  with `.empty`
- replace all use of  `TestState.userIdentifier` with `"alice"`
- replace all use of  `TestState.userList` with `.empty` except for `test_dispatch_withActionThatDoesChangeInternalState_delegateTriggered` where you'll need to use the value directly.

I've actually done all this for you on this branch...
https://github.com/pusher/chatkit-swift/tree/OKP/Reducers-remove-TestState

Reply:
In one of my previous PRs you asked me to centralize test data and now I am supposed to do the opposite. Sure...why not...

Reply:
I don't recall making such a request.  Are you are to direct me to it please?

Reply:
> Do you think we should centralised the value "joe"? Happy for this to be done in another PR though.

> I think you added TestUser.joe but you forgot to consume it here :-)

https://github.com/pusher/chatkit-swift/pull/246

Reply:
Just revisiting this after my holidays.  Regarding the quoted comment (https://github.com/pusher/chatkit-swift/pull/246#discussion_r370168330), I think you have misinterpreted or taken things out of context.  The comment relates to a value in `Demo/Controllers/RoomListViewController.swift`, a UIViewController of the Demo app, and therefore is of no relevance to **test** data.